components/transport-jms/src/test/java/org/wso2/transport/jms/test/client/polling/QueueClientPollingWithSelectorTestCase.java

```diff
@@ -0,0 +1,185 @@
+        // this should receive 10 messages but loop will goon for two more iterations to see if any more messages comes
+        for (int i = 0; i < 12; i++) {
+            Message message = jmsClientConnectorQueue
+                    .poll(queueName, 500, "JMSCorrelationID = '" + correlationId1 + "'");
+            if (message != null) {
+                receivedMsgCount1++;
+            }
+        }
+
+        int receivedMsgCount2 = 0;
+        // this should receive 5 messages but loop will goon for two more iterations to see if any more messages comes
+        for (int i = 0; i < 7; i++) {
+            Message message = jmsClientConnectorQueue
+                    .poll(queueName, 500, "JMSCorrelationID = '" + correlationId2 + "'");
+            if (message != null) {
+                receivedMsgCount2++;
+            }
+        }
+
+        JMSTestUtils.closeResources((JMSClientConnectorImpl) jmsClientConnectorQueue);
+
+        Assert.assertEquals(receivedMsgCount1, 10,
+                "JMS Client selector polling receiver one expected message count " + 10 + " , received message count "
+                        + receivedMsgCount1);
+        Assert.assertEquals(receivedMsgCount2, 5,
+                "JMS Client selector polling receiver two expected message count " + 5 + " , received message count "
+                        + receivedMsgCount2);
+    }
+
+    @AfterClass(groups = "jmsPolling",
+                description = "Shutting down the server by closing the only connection")
+    public void cleanup() throws JMSException {
+        if (connection != null) {
+            connection.close();
+        }
+    }
+
+    /**
+     * Publish messages to a queue.
+     *
+     * @param correlationId correlationId of the message.
+     * @param count number of messages.
+     * @throws JMSException thrown if error in jms resource creation or publishing.
+     * @throws InterruptedException thrown if interruption while sleeping.
+     */
+    private void publishToQueue(String correlationId, int count) throws JMSException {
+        Connection connection = null;
+        Session session = null;
+        MessageProducer producer = null;
+        try {
+            connection = jmsServer.createConnection(null, null);
+            session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+
+            producer = session.createProducer(session.createQueue(queueName));
+            producer.setDeliveryMode(DeliveryMode.NON_PERSISTENT);
+
+            logger.info("Publishing " + count + " messages to the queue: " + queueName);
+            for (int index = 0; index < count; index++) {
+                String queueText = "Queue Message : " + (index + 1);
+                TextMessage queueMessage = session.createTextMessage(queueText);
+                queueMessage.setJMSCorrelationID(correlationId);
+                producer.send(queueMessage);
+            }
+        } catch (JMSException e) {
+            throw e;
```

Comment:
Shall we remove this catch block since we do nothing here, simply just throwing again

Reply:
fixed,